src/main/java/it/gov/pagopa/paymentupdater/service/PaymentServiceImpl.java

```diff
@@ -70,52 +75,61 @@ public Optional<Payment> getPaymentByNoticeNumberAndFiscalCode(String noticeNumb
 	public void save(Payment reminder) {
 		paymentRepository.save(reminder);
 		log.info("Saved payment id: {}", reminder.getId());
-	}
+	} 
 
 	@Override
-	public Map<String, Boolean> checkPayment(Payment payment)
-			throws JsonProcessingException, InterruptedException, ExecutionException {
-		Map<String, Boolean> map = new HashMap<>();
-		map.put(isPaid, false);
-		try {
+	public Map<String, String> checkPayment(Payment payment) throws JsonProcessingException, InterruptedException, ExecutionException {
+		Map<String, String> map = new HashMap<>();
+		map.put(isPaid, "false");
+		try {			
 			ApiClient apiClient = new ApiClient();
 			if (enableRestKey) {
 				apiClient.addDefaultHeader("Ocp-Apim-Subscription-Key", proxyEndpointKey);
 			}
 			apiClient.setBasePath(urlProxy);
-
-			DefaultApi defaultApi = new DefaultApi();
-			defaultApi.setApiClient(apiClient);
-			defaultApi.getPaymentInfo(payment.getRptId(), Constants.X_CLIENT_ID);
+			
+			defaultApi.setApiClient(apiClient);	
+			PaymentRequestsGetResponse resp = defaultApi.getPaymentInfo(payment.getRptId(), Constants.X_CLIENT_ID);	
+			map.put("dueDate", resp.getDueDate());
 
 			return map;
+
+
 		} catch (HttpServerErrorException errorException) {
 			// the reminder is already paid
-			ProxyPaymentResponse res = mapper.readValue(errorException.getResponseBodyAsString(),
-					ProxyPaymentResponse.class);
-			if (!StringUtils.isEmpty(res.getDetail_v2())) {
-				if (res.getDetail_v2().equals("PPT_RPT_DUPLICATA")
-						&& errorException.getStatusCode().equals(HttpStatus.INTERNAL_SERVER_ERROR)) {
+			ProxyPaymentResponse res = mapper.readValue(errorException.getResponseBodyAsString(), ProxyPaymentResponse.class);
+			if (res.getDetail_v2().equals("PPT_RPT_DUPLICATA")
+					&& errorException.getStatusCode().equals(HttpStatus.INTERNAL_SERVER_ERROR)) {
 					payment.setPaidFlag(true);
 					payment.setPaidDate(LocalDateTime.now());
 					PaymentMessage message = new PaymentMessage();
 					message.setNoticeNumber(payment.getContent_paymentData_noticeNumber());
 					message.setPayeeFiscalCode(payment.getContent_paymentData_payeeFiscalCode());
 					message.setSource("payments");
+								
+					if(StringUtils.isNotEmpty(res.getDuedate())) {
+						LocalDate localDateDueDate = LocalDate.parse(res.getDuedate());
+				
+						LocalDate reminderDueDate = payment.getDueDate() != null ? payment.getDueDate().toLocalDate() : null;
```

Comment:
Try to use Optional :)

Reply:
This has been fixed in the develop-rptid branch that will be merged later